### PR TITLE
fix: manage the time when a ping happens to avoid the fixed yaw issue

### PR DESCRIPTION
**Pull request by kapeps**

### CODE REVIEW

File: tasks/Task.cpp [hunk 1]
```diff
@@ -92,10 +90,20 @@ bool Task::configureHook()
...
-
-    if (!TaskBase::configureHook()) {
+    m_orientation_output_flag = _orientation_output_flag.get();
+    m_ping_refresh_period = _ping_refresh_period.get();
+
+    if (m_orientation_output_flag &&
+        MINIMUM_PING_STATUS_REFRESH_TIME > m_ping_refresh_period) {
+        LOG_ERROR_S << "If an orientation is required, because of hardware limitations, "
+                       "the minimum ping refresh time is : "
+                    << MINIMUM_PING_STATUS_REFRESH_TIME << std::endl;
```
Comment (DeboraTahara) [inline tasks/Task.cpp]: Change the message:
Due to hardware limitations, the minimum ping update rate required to get orientation is:

File: tasks/Task.cpp [hunk 2]
```diff
@@ -131,16 +139,28 @@ bool Task::startHook()
 void Task::updateHook()
 {
     Status status = mDriver->autoStatus();
-    auto rbs_reference = convertToZWithOrientationRBS(status);
+    RigidBodyState rbs_reference;
+    // Write the local usbl depth
+    rbs_reference.time = status.timestamp;
```
Comment (DeboraTahara) [inline tasks/Task.cpp]: Timestamp it after get the position and orientation

File: tasks/Task.hpp [hunk 3]
```diff
@@ -8,6 +8,12 @@
 #include <base/samples/RigidBodyState.hpp>
 #include <usbl_seatrac/Driver.hpp>
 
+/**
+ * The minimum ping refresh rate in case it will need the local usbl's orientation
+ * information
+ */
+static const base::Time MINIMUM_PING_STATUS_REFRESH_TIME = base::Time::fromSeconds(2.4);
+
```
Comment (DeboraTahara) [inline tasks/Task.hpp:16]: How did you get 2.4?

**Review by DeboraTahara — APPROVED** [review]

File: tasks/Task.hpp (hunk 3 above)
Comment (kapeps) [reply tasks/Task.hpp:16]: I tested it in field. its the minimum refresh time possible that allows a correct orientation to be extracted from the ping.

File: usbl_seatrac.orogen [hunk 4]
```diff
@@ -54,8 +54,12 @@ task_context "Task", subclasses: "iodrivers_base::Task" do
     # Maximum time that a beacon is not communicated with before its position filter
     # is reset
     property "xcvr_posflt_tmo", "base/Time"
+    # Ping's minimum refresh_period
+    property "ping_refresh_period", "base/Time"
```
Comment (DeboraTahara) [inline usbl_seatrac.orogen:58]: I guess ping_update_rate  is a better name

**Review by DeboraTahara — APPROVED** [review]

File: usbl_seatrac.orogen (hunk 4 above)
Comment (kapeps) [reply usbl_seatrac.orogen:58]: its not the rate, its the period. maybe ping_update_period?

**Pull request merged**